indigo_app/views/works.py

```diff
@@ -670,123 +673,154 @@ def get_works(self, table, form):
 
     def link_repeal(self, info, form):
         # if the work is `repealed_by` something, try linking it
         # make a task if this fails
         # (either because the work isn't found or because the repeal date isn't right,
         # which could be because it doesn't exist or because it's in the wrong format)
-        if info.get('repealed_by'):
-            try:
-                repealing_work = self.find_work_by_title(info.get('repealed_by'))
-            except Work.DoesNotExist:
-                return self.create_task(info, form, task_type='repeal')
-
-            repeal_date = repealing_work.commencement_date
-            repeal_date = info.get('with_effect_from') or repeal_date
-
-            if not repeal_date:
-                return self.create_task(info, form, task_type='repeal')
+        work = info['work']
+        repealing_work = self.find_work_by_title(info['repealed_by'])
+        if not repealing_work:
+            return self.create_task(info, form, task_type='repeal')
 
-            info['work'].repealed_by = repealing_work
-            info['work'].repealed_date = repeal_date
+        repeal_date = repealing_work.commencement_date
+        if not repeal_date:
+            return self.create_task(info, form, task_type='repeal')
 
-            try:
-                info['work'].save_with_revision(self.request.user)
-            except ValidationError:
-                self.create_task(info, form, task_type='repeal')
+        work.repealed_by = repealing_work
+        work.repealed_date = repeal_date
 
-    def find_work_by_title(self, title):
-        return Work.objects.get(title=title, country=self.country, locality=self.locality)
+        try:
+            work.save_with_revision(self.request.user)
+        except ValidationError:
+            self.create_task(info, form, task_type='repeal')
 
     def link_parent_work(self, info, form):
         # if the work has a `parent_work`, try linking it
         # make a task if this fails
-        if info.get('parent_work'):
-            try:
-                info['work'].parent_work = self.find_work_by_title(info.get('parent_work'))
-                info['work'].save_with_revision(self.request.user)
-            except Work.DoesNotExist:
-                self.create_task(info, form, task_type='parent_work')
+        work = info['work']
+        parent_work = self.find_work_by_title(info['parent_work'])
+        if not parent_work:
+            return self.create_task(info, form, task_type='parent_work')
+
+        work.parent_work = self.find_work_by_title(info.get('parent_work'))
+
+        try:
+            work.save_with_revision(self.request.user)
+        except ValidationError:
+            self.create_task(info, form, task_type='parent_work')
+
+    def find_work_by_title(self, title):
+        potential_match = Work.objects.filter(title=title, country=self.country, locality=self.locality)
+        if len(potential_match) == 1:
+            return potential_match
```

Comment:
This returns the queryset, not the actual object. Use `.filter(...).first()` to get the first matching work.